vscode/microsoft-kiota/src/openApiTreeProvider.ts

```diff
@@ -13,47 +24,83 @@ export class OpenApiTreeProvider implements vscode.TreeDataProvider<OpenApiTreeN
-      if (this._lockFile?.descriptionLocation) {
-        this._descriptionUrl = this._lockFile.descriptionLocation;
-        this.includeFilters = this._lockFile.includePatterns;
-        this.excludeFilters = this._lockFile.excludePatterns;
-        const settings = this.settingsGetter();
-        await this.loadNodes(settings.clearCache);
-        if (this.rawRootNode) {
-            this.refreshView();
+    public async loadLockFile(path: string, clientOrPluginName?: string): Promise<void> {
+        this.closeDescription(false);
+        this._lockFilePath = path;
+        const lockFileData = await vscode.workspace.fs.readFile(vscode.Uri.file(path));
+        let parsedLockFile = JSON.parse(lockFileData.toString()) as LockFile;
+
+        if (clientOrPluginName) {
+            let filteredData: Partial<LockFile> = { version: parsedLockFile.version };
+
+            if (parsedLockFile.clients && parsedLockFile.clients[clientOrPluginName]) {
+                filteredData.clients = {
+                    [clientOrPluginName]: parsedLockFile.clients[clientOrPluginName]
+                };
+            }
+
+            if (parsedLockFile.plugins && parsedLockFile.plugins[clientOrPluginName]) {
+                filteredData.plugins = {
+                    [clientOrPluginName]: parsedLockFile.plugins[clientOrPluginName]
+                };
+            }
+
+            parsedLockFile = filteredData as LockFile;
+        }
+
+        this._lockFile = parsedLockFile;
+
+        const clientOrPlugin: ClientOrPluginProperties | undefined =
+            Object.values(this._lockFile.clients ?? {})[0] ||
+            Object.values(this._lockFile.plugins ?? {})[0];
+
+        if (clientOrPlugin) {
+            this._descriptionUrl = clientOrPlugin.descriptionLocation;
+            this.includeFilters = clientOrPlugin.includePatterns;
+            this.excludeFilters = clientOrPlugin.excludePatterns;
+
+            const settings = this.settingsGetter();
+            await this.loadNodes(settings.clearCache, clientOrPluginName);
+
+            if (this.rawRootNode) {
+                this.refreshView();
+            }
         }
-      }
     }
-    public async loadEditPaths(clientObject: any): Promise<void> {
+    public async loadEditPaths(clientOrPluginKey: string, clientObject: ClientOrPluginProperties): Promise<void> {
         this.closeDescription(false);
-        this._lockFile = clientObject;
-        if (this._lockFile?.descriptionLocation) {
-          this._descriptionUrl = this._lockFile.descriptionLocation;
-          this.includeFilters = this._lockFile.includePatterns;
-          this.excludeFilters = this._lockFile.excludePatterns;
-          const settings = this.settingsGetter();
-          await this.loadNodes(settings.clearCache);
-          if (this.rawRootNode) {
-              this.refreshView();
-          }
+        this._lockFile = { version: '1.0.0', clients: {}, plugins: {} };
```

Comment:
I'd declare a constant here instead, and assign the field with the constant value after the if block. Or something along these lines. This way you don't need the bangs anymore in the lines below

Reply:
Okay yeah!